torch/csrc/jit/codegen/cuda/partition.cpp

```diff
@@ -52,10 +52,52 @@ static bool isFusableDevice(const Node* node) {
   return device->is_cuda();
 }
 
+inline bool isFuserSupportedType(
+    const std::shared_ptr<const TensorType>& tensor) {
+  auto scalar_type = tensor->scalarType();
+  if (scalar_type.has_value()) {
+    switch (scalar_type.value()) {
+      // following aten_to_data_type interface
+      case at::ScalarType::Bool:
+      case at::ScalarType::Float:
+      case at::ScalarType::Half:
+      case at::ScalarType::Long:
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  // Still allowing NULL type, should we prevent fusing those?
+  return true;
+}
+
+inline bool isNodeScalarDataTypeCompatible(const Node* node) {
+  for (auto output : node->outputs()) {
+    if (auto out_type = output->type()->cast<const TensorType>()) {
```

Comment:
nitpick: output/input check are identical, we can lift them to the same function to avoid boiler plate.